Source/WebCore/page/NavigatorLoginStatus.h

```diff
@@ -26,31 +26,34 @@
 #pragma once
 
 #include "Supplementable.h"
+#include <wtf/WeakPtr.h>
 
 namespace WebCore {
 
 class DeferredPromise;
 class Navigator;
-class NavigatorLoginStatus final : public Supplement<Navigator> {
+class WeakPtrImplWithEventTargetData;
+enum class IsLoggedIn : uint8_t;
+
+class NavigatorLoginStatus final : public Supplement<Navigator>, public CanMakeWeakPtr<NavigatorLoginStatus> {
     WTF_MAKE_FAST_ALLOCATED;
 public:
     explicit NavigatorLoginStatus(Navigator& navigator)
         : m_navigator(navigator)
     {
     }
-    static void setLoggedIn(Navigator&, Ref<DeferredPromise>&&);
-    static void setLoggedOut(Navigator&, Ref<DeferredPromise>&&);
+    static void setLoginStatus(Navigator&, const String& status, Ref<DeferredPromise>&&);
     static void isLoggedIn(Navigator&, Ref<DeferredPromise>&&);
 
 private:
-    void setLoggedIn(Ref<DeferredPromise>&&);
-    void setLoggedOut(Ref<DeferredPromise>&&);
+    void setLoginStatus(const String& status, Ref<DeferredPromise>&&);
     void isLoggedIn(Ref<DeferredPromise>&&);
 
     static NavigatorLoginStatus* from(Navigator&);
     static ASCIILiteral supplementName();
 
     Navigator& m_navigator;
+    IsLoggedIn m_loginStatus;
```

Comment:
I think it would be best to remove this and only keep this state in the network process.

Reply:
Initially we need to know what's the status of login. So we get the status as a string parameter:
https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLogin/setStatus

Reply:
I need at least a variable to keep the status when deciding about whether it is 'logged-in' or 'logged-out'. Does that work if I define it before he if else statement?
```
IsLoggedIn loginStatus;
    if (status == WTF::String::fromLatin1("logged-in"))
        loginStatus = IsLoggedIn::Yes;
    else if (status == WTF::String::fromLatin1("logged-out"))
        loginStatus = IsLoggedIn::No;
    else {
        promise->reject();
        return;
    }
```

Reply:
Yes, a local variable should be fine.